File: Automacao.DuamApi.Web/src/types/job.ts
export type JobStatus = "Pendente" | "EmProcessamento" | "Concluido" | "Falhou";

export const TERMINAL_STATUSES: JobStatus[] = ["Concluido", "Falhou"];

export interface ResultadoLinhaResponse {
  linha: number;
  inscricao: string;
  sucesso: boolean;
  mensagem: string;
  dataProcessamento: string;
}

export interface JobStatusResponse {
  jobId: string;
  status: JobStatus;
  usuario: string;
  nomeArquivoOriginal: string;
  totalLinhas: number | null;
  linhasProcessadas: number;
  linhasComErro: number;
  dataCriacao: string;
  dataInicio: string | null;
  dataFim: string | null;
  mensagemErro: string | null;
  resultados: ResultadoLinhaResponse[];
}

export interface JobResumoResponse {
  jobId: string;
  status: JobStatus;
  usuario: string;
  nomeArquivoOriginal: string;
  totalLinhas: number | null;
  linhasProcessadas: number;
  linhasComErro: number;
  dataCriacao: string;
  dataInicio: string | null;
  dataFim: string | null;
}

export interface JobListaResponse {
  total: number;
  pagina: number;
  tamanhoPagina: number;
  jobs: JobResumoResponse[];
}

export interface JobEnqueuedResponse {
  jobId: string;
  status: JobStatus;
  statusUrl: string;
}

export interface JobListFilters {
  usuario?: string;
  status?: JobStatus | "";
  pagina?: number;
  tamanhoPagina?: number;
}

File: Automacao.DuamApi/Automacao.DuamApi/Models/Job.cs
namespace DuamApi.Models;

public enum JobStatus
{
    Pendente,
    EmProcessamento,
    Concluido,
    Falhou
}

public enum TipoInscricao
{
    Imobiliaria,
    Economica
}

public class Job
{
    public Guid Id { get; set; }

    public string Usuario { get; set; } = string.Empty;

    public TipoInscricao TipoInscricao { get; set; }

    public string NomeArquivoOriginal { get; set; } = string.Empty;

    public string CaminhoArquivo { get; set; } = string.Empty;

    public JobStatus Status { get; set; } = JobStatus.Pendente;

    public int? TotalLinhas { get; set; }

    public int LinhasProcessadas { get; set; }

    public string? MensagemErro { get; set; }

    public DateTime DataCriacao { get; set; }

    public DateTime? DataInicio { get; set; }

    public DateTime? DataFim { get; set; }

    public List<ResultadoLinhaRegistro> Resultados { get; set; } = new();
}
